lib/bloc_record/persistence.rb

```diff
@@ -21,7 +21,110 @@ def create(attrs)
       data["id"] = connection.execute("SELECT last_insert_rowid();")[0][0]
       new(data)
     end
+
+
+    def update(ids, updates)
+      if updates.keys.uniq == updates.keys
+        updates = BlocRecord::Utility.convert_keys(updates)
+        updates.delete "id"
+        updates_array = updates.map { |key, value| "#{key}=#{BlocRecord::Utility.sql_strings(value)}" }
+
+        if ids.class == Fixnum
+          where_clause = "WHERE id = #{ids};"
+        elsif ids.class == Array
+          where_clause = ids.empty? ? ";" : "WHERE id IN (#{ids.join(",")});"
+        else
+          where_clause = ";"
+        end
+
+        connection.execute <<-SQL
+          UPDATE #{table}
+          SET #{updates_array * ","} #{where_clause}
+        SQL
+
+        true
+      else
+        updates.delete "id"
+        keys = updates.keys
+        potential_duplicate = nil
+        keys.each_with_index do |name, index|
+          if potential_duplicate == name
+            duplicate = name
+            duplicate_index = index
+            break
+          else
+            potential_duplicate = name
+          end
+        end
+        new_hash = updates[duplicate]
+        new_id = ids[duplicate_index]
+        updates.delete(duplicate)
+        ids.delete(duplicate_index)
+        update(new_id, new_hash)
+        update(ids, updates)
+      end
+    end
+
+    def update_all(updates)
+      update(nil, updates)
+    end
+
+    def destroy(*id)
+      if id.length > 1
+        where_clause = "WHERE id IN (#{id.join(",")});"
+      else
+        where_clause = "WHERE id = #{id.first};"
+      end
+      connection.execute <<-SQL
+        DELETE FROM #{table} #{where_clause}
+      SQL
+
+      true
+    end
+
+    def destroy_all(conditions_hash=nil, *extra_args)
+      if conditions_hash && !conditions_hash.empty?
+        case conditions_hash
+        when Hash
+          conditions_hash = BlocRecord::Utility.convert_keys(conditions_hash)
+          conditions = conditions_hash.map{|key, value| "#{key}=#{BlocRecord::Utility.sql_strings(value)}"}.join(" and ")
+        when String
+          if extra_args.is_a?(Array)
+            conditions_array = conditions_hash.split("?")
+            conditions_array = conditions_array.each_with_index.map do |condition, index|
+              condition + "'" + extra_args[index] + "'"
```

Comment:
One thing I just considered: you should be cleaning the `extra_args[index]` value with the `sql_strings` method used above, since that is the point of doing things this way. I will not require a fix to pass, though, since the syntax is what you were supporting and nothing states you should be cleaning the value.